### DISCUSSION TITLE
Skeleton v3 Pre-Release Announcement 🎉

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#2681] [Announcements]
![announcement](https://github.com/skeletonlabs/skeleton/assets/1509726/5dac70de-857f-4721-bf3a-65cb5669f867)

Hey everyone, Chris here from Skeleton. As you may know, over the last few months we've been hard at work on [Skeleton v3](https://github.com/skeletonlabs/skeleton/discussions/2375). As of today, we're thrilled to share that we've now reached a major milestone - Skeleton v3 is now available in early access via a new public pre-release! This means you can now install and try all the new features directly!

## View the Documentation

We've provide a new documentation website here: https://next.skeleton.dev/

## Progress Updates

As we continue forward, we'll update the section below with updates.

View our [YouTube playlist of progress updates](https://www.youtube.com/watch?v=auFvaLyN9PQ&list=PLJtgPDPrJYAv8fn5gM1J5XDlL2VkHqJPB) or review our written updates below.

- [Pre-Release Walkthrough (Video)](https://youtu.be/oEFrp1cwubE)
- https://github.com/skeletonlabs/skeleton/discussions/2424
- https://github.com/skeletonlabs/skeleton/discussions/2452
- https://github.com/skeletonlabs/skeleton/discussions/2477
- https://github.com/skeletonlabs/skeleton/discussions/2521
- https://github.com/skeletonlabs/skeleton/discussions/2544
- https://github.com/skeletonlabs/skeleton/discussions/2574
- https://github.com/skeletonlabs/skeleton/discussions/2602
- https://github.com/skeletonlabs/skeleton/discussions/2629
- https://github.com/skeletonlabs/skeleton/discussions/2652
- https://github.com/skeletonlabs/skeleton/discussions/2681
- [Skeleton Next - Update 9 - Doc Search, Gradients, and Switch (Video)](https://youtu.be/awYCE9L_tkc)
- https://github.com/skeletonlabs/skeleton/discussions/2728
- https://github.com/skeletonlabs/skeleton/discussions/2784
- https://github.com/skeletonlabs/skeleton/discussions/2804
- https://github.com/skeletonlabs/skeleton/discussions/2861
- https://github.com/skeletonlabs/skeleton/discussions/2871
- https://github.com/skeletonlabs/skeleton/discussions/2885
- https://github.com/skeletonlabs/skeleton/discussions/2886

## How to Install

> [!CAUTION]
> Skeleton v3 is available in a very early pre-release, meaning you can and will encounter bugs, missing features, or incomplete documentation. It is not advised to use in production. Please proceed with caution.

Given the early status of the pre-release, we're going to ask that you limit testing to _new projects_ for now. We've provided guide instruction for each supported framework below:

- [Install for React + NextJS](https://next.skeleton.dev/docs/get-started/installation/nextjs)
- [Install for Svelte 5 RC + SvelteKit](https://next.skeleton.dev/docs/get-started/installation/sveltekit)

> NOTE: keep and eye out for additional installation guides over the next few weeks!

## What's New

Skeleton v3 represents a completely top-to-bottom rewrite of most features. Let's review a few of the most exciting changes:

- Skeleton now features a modular structure:
  - The framework agnostic core package provides the [Tailwind Plugin](https://next.skeleton.dev/docs/get-started/plugin) (design system, themes, and more)
  - Dedicated framework-specific component packages are available both React and Svelte 5
  - Plus direct support for add-ons such as the upcoming [Floating UI Svelte](https://floating-ui-svelte.vercel.app/).
- We now provide a much more robust design system, including:
  - A new design aesthetic and several new curated preset themes
  - A new and improved [theme system](https://next.skeleton.dev/docs/design/themes) and [generator](https://themes.skeleton.dev/).
  - An [expanded color system](https://next.skeleton.dev/docs/design/colors) featuring contrasting tones, support for shade 950, and improved color pairings
  - New [Presets system](https://next.skeleton.dev/docs/design/presets) for handling common reused styles, with automatic support for contrasting colors
  - Improved [Typography system](https://next.skeleton.dev/docs/design/typography), including theme-specific scaling, and (soon) semantic text styles
  - The new [Spacing system](https://next.skeleton.dev/docs/design/spacing), enabling dynamic scaling based on a theme-specified scale factor.
  - Iconography remains an agnostic choice, while providing first-class support for [Lucide](https://lucide.dev/).
- We've completely revitalized a number of features:
  - Tailwind elements are now easier to use and more flexible when combined
  - We now provide a full-fledge set of React components that adhere to Skeleton conventions
  - Svelte components have been overhauled to take advantage of Svelte 5 features like runes, snippets, and more
- A new documentation experience:
  - [Simpler onboarding](https://next.skeleton.dev/docs/get-started/installation) to help get you started quickly
  - More in-depth exploration of the [fundamentals of Skeleton](https://next.skeleton.dev/docs/get-started/fundamentals)
  - We now provide complete code examples showcasing the entire process for implementing each feature.
  - Added a new [Cookbook](https://next.skeleton.dev/docs/resources/cookbook) section, showcasing simple recipes for mixing Skeleton + Tailwind features
  - Provide quick toggles between component frameworks, with detailed API reference at a glance

Plus much more! Expect a complete list upon release!

## Noteworthy Changes from v2

Below is a list of a few changes to be aware of if you're coming from Skeleton v2:

- We do not yet provide a v2 to v3 migration guide - but will have one before our stable release!
- v2 themes are currently incompatible with v3 themes - consider generating a new theme instead.
- We provide a limited set of [built-in v3 themes](https://next.skeleton.dev/docs/design/themes) - but expect more in the future!
- We're still in the process of porting many of our components to v3 - expect regular updates to this!
- There are a number of breaking changes to component features due to Svelte 5 and our [new approach to styles](https://next.skeleton.dev/docs/get-started/fundamentals#style-props).
- [v2 variants classes](https://www.skeleton.dev/docs/variants) have now been replaced by the much more robust [Presets system](https://next.skeleton.dev/docs/design/presets).
- [v2 design token classes](https://www.skeleton.dev/docs/tokens) have been replaced by our new extended [Tailwind utility classes](https://next.skeleton.dev/docs/get-started/plugin).
- [Component API Reference](https://next.skeleton.dev/docs/components/accordion/react#api-reference) is only available for React while we await Svelte 5 to reach a stable release.

## Providing Feedback

Our next step is to gather feedback from you, our community. This might come in the form of feedback and comments in regards to any of the following:

- What changes you do or do not like
- Features that don't meet expectations (what works, what doesn't)
- Gaps and errors in the documentation
- Proposals for quality improvements, especially around our React implementation
- And anything else you wish to share!

The team and I have put a ton of blood, sweat, and tears into this project and are very excited to hear your feedback. We just ask that you please try to keep your feedback kind and constructive!

You may contact us in the the comment section below, on GitHub via either [new issues](https://github.com/skeletonlabs/skeleton/issues/new/choose) or within [existing milestone issues](https://github.com/skeletonlabs/skeleton/milestone/5), and of course by [Discord](https://discord.gg/EXqV7W8MtY) in the dedicated `#skeleton-next` channel.

Thanks, and please enjoy!

**Comment by rChaoz** [comment]
One small thing - when viewing the docs on mobile, there's no way to view anything else other than the introduction page. Even when enabling desktop mode in the browser in landscape. There's no sidebar or anything to navigate.

> Moderator Note: (image removed due to size)

**Reply by endigo9740** [reply]
@rChaoz oh yeah, we're aware. Lots of work left on the doc site before launch, but we're actually waiting on our upcoming Floating UI Svelte package to handle this!

- https://github.com/skeletonlabs/floating-ui-svelte/discussions/50

You'll likely see this come in alongside our upcoming integration guide on the v3 doc site. It'll give us a great way to dogfood the new library! Feel free to report any similar issues here:

- https://github.com/skeletonlabs/skeleton/issues/2353

**Comment by mmailaender** [comment]
Typography Ordered is an unordered list
https://next.skeleton.dev/docs/design/typography#ordered 
![image](https://github.com/skeletonlabs/skeleton/assets/87228994/f4277d9e-b3ba-4994-b2c6-86b4697306ab)

**Comment by mmailaender** [comment]
For consistency, I think it makes sense to unify the import statement for react and svelte in iconography: https://next.skeleton.dev/docs/design/iconography
![image](https://github.com/skeletonlabs/skeleton/assets/87228994/94a1d72b-bd70-4145-8563-66b98b9d53dd)
=> `import { AlterCircle } from 'lucide-svelte';` (https://lucide.dev/guide/packages/lucide-svelte#example)

**Reply by Hugos68** [reply]
This is done for technical reasons. Importing from the root of the package causes Vite (what we use to develop) to load in *all* icons causing a 20 second delay everytime we start the development server.

**Reply by endigo9740** [reply]
What Hugo said, the direct import method is required for Svelte for proper tree shaking.
